tfx/components/example_gen/base_example_gen_executor.py

```diff
@@ -33,19 +34,24 @@
 
 
 def _GeneratePartitionKey(record: Union[tf.train.Example,
-                                        tf.train.SequenceExample, bytes],
+                                        tf.train.SequenceExample, bytes,
+                                        Dict[str, Any]],
                           split_config: example_gen_pb2.SplitConfig) -> bytes:
   """Generates key for partition."""
 
   if not split_config.HasField('partition_feature_name'):
     if isinstance(record, bytes):
       return record
+    if isinstance(record, dict):
+      return json.dumps(record).encode('utf-8')
```

Comment:
there might be a better way to encode the dictionaries as bytes when there is no partition feature name provided

Reply:
let's use json_utils (from tfx.utils import json_utils)